asciidoctorj-core/src/main/java/org/asciidoctor/cli/AsciidoctorInvoker.java

```diff
@@ -100,13 +103,37 @@ private void setVerboseLevel(AsciidoctorCliOptions asciidoctorCliOptions) {
     }
 
     private Asciidoctor buildAsciidoctorJInstance(AsciidoctorCliOptions asciidoctorCliOptions) {
-        Asciidoctor asciidoctor;
-        if(asciidoctorCliOptions.isLoadPaths()) {
-         asciidoctor = JRubyAsciidoctor.create(asciidoctorCliOptions.getLoadPaths());   
-        } else {
-            asciidoctor = JRubyAsciidoctor.create();
+        ClassLoader oldTccl = Thread.currentThread().getContextClassLoader();
+        try {
+            if (asciidoctorCliOptions.isClassPaths()) {
+                URLClassLoader tccl = createUrlClassLoader(asciidoctorCliOptions.getClassPaths());
+                Thread.currentThread().setContextClassLoader(tccl);
+            }
+            Asciidoctor asciidoctor;
+            if (asciidoctorCliOptions.isLoadPaths()) {
+                asciidoctor = JRubyAsciidoctor.create(asciidoctorCliOptions.getLoadPaths());
+            } else {
+                asciidoctor = JRubyAsciidoctor.create();
+            }
+            return asciidoctor;
+        } finally {
+            Thread.currentThread().setContextClassLoader(oldTccl);
+        }
+    }
+
+    private URLClassLoader createUrlClassLoader(List<String> classPaths) {
```

Comment:
`--cp mydir` will load class files directly from `mydir`
`--cp mydir/*` will load classes from archives that are in `mydir` like `mydir/myconverter.jar`
`--cp mydir/myconverter.jar` will load classes from exactly that jar